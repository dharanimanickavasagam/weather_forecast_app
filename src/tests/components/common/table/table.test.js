import React from "react";
import { shallow } from "enzyme";
import Table from "../../../../components/common/table";

let mockColumns, mockData, wrapper;

beforeEach(() => {
  mockColumns = ["Pressure", "Humidity", "Min °F", "Max °F"];
  mockData = [{ pressure: 1017, humidity: 87, min_temp: 46, max_temp: 52 }];
  wrapper = shallow(<Table columns={mockColumns} data={mockData} />);
});

describe("it renders component Table", () => {
  test("it should render the component correctly with the passed props", () => {
    expect(wrapper.find("TableHeader").prop("columns")).toEqual(mockColumns);
    expect(wrapper.find("TableBody").prop("data")).toBe(mockData);
    expect(wrapper).toMatchSnapshot();
  });
});
