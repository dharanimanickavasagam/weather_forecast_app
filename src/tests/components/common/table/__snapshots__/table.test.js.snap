// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders component Table it should render the component correctly with the passed props 1`] = `
<table
  className="table table-hover text-center"
>
  <TableHeader
    columns={
      Array [
        "Pressure",
        "Humidity",
        "Min °F",
        "Max °F",
      ]
    }
  />
  <TableBody
    data={
      Array [
        Object {
          "humidity": 87,
          "max_temp": 52,
          "min_temp": 46,
          "pressure": 1017,
        },
      ]
    }
  />
</table>
`;
